cryptol-remote-api/README.rst

```diff
@@ -0,0 +1,190 @@
+    # part of the Python docstring.
+    In [5]: ctx.add?
+    Signature:   ctx.add(*args)
+    Type:        CryptolFunctionHandle
+    String form: <cryptol.CryptolFunctionHandle object at 0x7fa16e878c50>
+    File:        ~/Projects/proto/proto/python/cryptol/__init__.py
+    Docstring:   Cryptol type: {a} (fin a) => [a] -> [a] -> [a]
+
+    # There are heuristic rules for converting Python data to the associated Cryptol
+    # data, taking the Cryptol type into account:
+    In [6]: ctx.add(bytes.fromhex('ff'), bytes.fromhex('01'))
+    Out[6]: b'\x00'
+
+    # Additionally, the current state of a connection can be used to construct a Python
+    # module from which Cryptol names can be imported directly:
+
+    In [7]: cryptol.add_cryptol_module('Foo', c)
+
+    In [8]: from Foo import *
+
+    # Because b'\2' is enough to solve the type variable a in add's type, the integer 2
+    # can be used as a bitvector. There is not yet a way to supply a explicitly.
+    In [9]: add(b'\2', 2)
+    Out[9]: b'\x04'
+
+    # Cryptol documentation is also carried over to Python, whether through a Context or
+    # through a module.
+    In [10]: ctx.carry?
+    Signature:   ctx.carry(*args)
+    Type:        CryptolFunctionHandle
+    String form: <cryptol.CryptolFunctionHandle object at 0x7fa16e87b748>
+    File:        ~/Projects/proto/proto/python/cryptol/__init__.py
+    Docstring:
+    Cryptol type: {n} (fin n) => [n] -> [n] -> Bit
+    Unsigned carry.  Returns true if the unsigned addition of the given
+    bitvector arguments would result in an unsigned overflow.
+
+    In [11]: carry?
+    Signature:   carry(*args)
+    Type:        CryptolFunctionHandle
+    String form: <cryptol.CryptolFunctionHandle object at 0x7fa16e7bb6a0>
+    File:        ~/Projects/proto/proto/python/cryptol/__init__.py
+    Docstring:
+    Cryptol type: {n} (fin n) => [n] -> [n] -> Bit
+    Unsigned carry.  Returns true if the unsigned addition of the given
+    bitvector arguments would result in an unsigned overflow.
+
+
+Emacs
+~~~~~
+
+There is a little test rig written in Emacs Lisp to automate the
+production of commands and log responses. Emacs was chosen because it
+makes it easy to run a subprocess and communicate with it over a pipe
+or socket --- don't expect fancy editor support for Cryptol or much
+ease of use from the integration. Note that these commands can be
+sensitive to the current working directory in Emacs. The Emacs test
+rig is also in the ``argo`` repository.
+
+There are two ways to use it: over stdio, or over a socket. The
+initial setup for both is the same:
+
+1. Launch emacs
+
+2. Open ``proto-test.el``
+
+3. Evaluate the buffer: ``M-x eval-buffer`` or on Spacemacs: ``, e b``
+
+To use the stdio version:
+
+1. ``M-x proto-test-start``
+
+2. At the prompt for ``Command:``, run the server with ``cabal v2-exec -v0
+   saw-remote-api`` or ``cabal v2-exec -v0 cryptol-remote-api``.
```

Comment:
I'm not sure if this is relevant on this line, but don't both server APIs need one of `http`, `socket`, or `stdio` following their name to run them?